cli/flox/src/utils/init/catalog_client.rs

```diff
@@ -35,40 +36,37 @@ pub fn init_catalog_client(config: &Config) -> Result<Client, anyhow::Error> {
         );
         Ok(MockClient::new(Some(path))?.into())
     } else {
-        let mut extra_headers: BTreeMap<String, String> = BTreeMap::new();
+        let mut client_config = CatalogClientConfig {
+            catalog_url: config
+                .flox
+                .catalog_url
+                .clone()
+                .unwrap_or_else(|| DEFAULT_CATALOG_URL.to_string()),
+            floxhub_token: config.flox.floxhub_token.clone(),
+            extra_headers: None,
+        };
 
         // If metrics are not disabled, pass along the metrics UUID so it can be
         // sent in catalog request headers, as well as the Sentry span info
         if !config.flox.disable_metrics {
-            extra_headers.insert(
+            let mut metrics_headers = BTreeMap::new();
```

Comment:
nit: you could do `let extra_headers = ...this whole block...` and then create the `client_config` at the end

Reply:
I like it! TY

Reply:
You could also use `(!config.flox.disable_metrics).then()` if you like less nesting. I kinda think that's less readable. Personal preference I guess 🤷